### PR TITLE
chore: Replace OptionButton with Switch

### PR DESCRIPTION
**Pull request by ianjon3s**
## What does this PR do?

Replaces OptionButton with Switch in all cases. Previously entered true/false states are upheld.

Also removes the OptionButton component and Storybook file. I've not created a story for Switch as it is not a public-facing component (OptionButton was previously used for radio buttons on the public service).

**Components affected:**

- [Checklist](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/YJog6SF0FJ/edit)
- [Upload and label](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/egatkqr0Uc/edit)
- [Number input](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/Ah6HMrAmnO/edit)
- [Notice](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/gf2BRZ8J28/edit)
- [Find property](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/ybUl93yAjl/edit)
- [Property information](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/Q0dYGuBG5y/edit)
- [Draw boundary](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/OpdRD6XFjj/edit)
- [Calculate](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/D00vhR6wPG/edit)
- [Pay](https://3287.planx.pizza/testing/option-button-switch/nodes/_root/nodes/iIPZydvo1y/edit)

### CODE REVIEW

**Comment by RODO94** [comment]
`bug`: the checklist switch didn't work when I initially loaded up the checklist but it does spin into action after I add an Option

https://github.com/theopensystemslab/planx-new/assets/138574807/1a6bd841-83f6-4a2d-b399-d23c9779b24f

**Review by RODO94 — COMMENTED** [review]
Everything worked for me on the pizza except that one wee bit on the Checklist component, although it is quite minor

**Review by DafyddLlyr — APPROVED** [review]
Looks good thanks, nice to have this consistency and to remove the old `OptionButton` 👍 

The bug mentioned above by @RODO94 exists already and is not introduced here. If you can fix it, that's great, if not - no sweat! Let's just open a GitHub issue so we can track its.

File: editor.planx.uk/src/@planx/components/Checklist/model.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ interface ChecklistExpandableProps {
 export const toggleExpandableChecklist = (
   checklist: ChecklistExpandableProps,
 ): ChecklistExpandableProps => {
-  if (checklist.options) {
+  if (checklist.options !== undefined && checklist.options.length > 0) {
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Checklist/model.ts:26]: Perfect - super quick fix! 💪

**Review by DafyddLlyr — APPROVED** [review]

**Comment by ianjon3s** [comment]
Thanks for the review @RODO94 @DafyddLlyr ,

I've managed to update the model so that the "expandable" option is available before an option is added. This seems a cleaner solution than disabling as it means lists can be built starting from section/group titles, rather than needing to add an option to begin.

**Pull request merged**